integration/container/exec_test.go

```diff
@@ -139,3 +142,73 @@ func TestExecUser(t *testing.T) {
 
 	assert.Assert(t, is.Contains(result.Stdout(), "uid=1(daemon) gid=1(daemon)"), "exec command not running as uid/gid 1")
 }
+
+// Borrowed from daemon/util_test.go
+type MockContainerdClient struct {
+}
+func (c *MockContainerdClient) CloseStdin(ctx context.Context, containerID, processID string) error {
+	return nil
+}
+
+
+type mockContainerd struct {
+	MockContainerdClient
+	calledCtx         *context.Context
+	calledContainerID *string
+	calledID          *string
+	calledSig         *int
+}
+
+func (cd *mockContainerd) SignalProcess(ctx context.Context, containerID, id string, sig int) error {
+	cd.calledCtx = &ctx
+	cd.calledContainerID = &containerID
+	cd.calledID = &id
+	cd.calledSig = &sig
+	return nil
+}
+
+func TestContainerExecKillNoSuchExec(t *testing.T) {
+	mock := mockContainerd{}
+	ctx := context.Background()
+	// d := &Daemon{
+	// 	execCommands: exec.NewStore(),
+	// 	containerd:   &mock,
+	// }
+	d := daemon.New(t)
```

Comment:
Hello @AkihiroSuda. I've done my best to try to fix the issues I found. On the good side: the unit tests now pass!

But the integration tests fail with:
```
# github.com/docker/docker/integration/container [github.com/docker/docker/integration/container.test]
integration/container/exec_test.go:179:10: d.ContainerExecKill undefined (type *daemon.Daemon has no field or method ContainerExecKill)
integration/container/exec_test.go:190:8: undefined: "github.com/docker/docker/integration/internal/container".Container
integration/container/exec_test.go:205:3: d.containers undefined (type *daemon.Daemon has no field or method containers)
integration/container/exec_test.go:206:3: d.registerExecCommand undefined (type *daemon.Daemon has no field or method registerExecCommand)
integration/container/exec_test.go:208:10: d.ContainerExecKill undefined (type *daemon.Daemon has no field or method ContainerExecKill)
Makefile:238: recipe for target 'test-integration' failed
```

And my lack of understanding of Go and the project itself make me unable to advance further. I hit a wall. Could you please give me a hand to figure out how to fix these tests?